### PR TITLE
feat(mls): create an mls group FS-540

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-540" title="FS-540" target="_blank">TaskFS-540</a>  [iOS] Create an MLS conversation
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- Update the `POST /conversations` payload to create mls groups.
- When processing the response, ask `MLSController` to create the mls group.
- Update Conversation payload to contain the `epoch` (currently not used yet).

### Dependencies

Needs releases with:

- https://github.com/wireapp/wire-ios-data-model/pull/1390

#### Test Coverage

- Test request generation
- Test response processing
- Test payload processing (store group id)

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by David-Henner — APPROVED** [review]

**Review by ghuMustafa — APPROVED** [review]

File: Sources/Payloads/Processing/PayloadProcessing+Conversation.swift [hunk 1]
```diff
@@ -168,6 +168,7 @@ extension Payload.Conversation {
         conversation.conversationType = .group
         conversation.remoteIdentifier = conversationID
         conversation.domain = APIVersion.isFederationEnabled ? qualifiedID?.domain : nil
+        conversation.mlsGroupID = mlsGroupID.flatMap(MLSGroupID.init(base64Encoded:))
```
Comment (David-Henner) [inline Sources/Payloads/Processing/PayloadProcessing+Conversation.swift:171]: Note: This already happens in `updateMLSStatus(for: conversation, context: context)`

File: Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift [hunk 2]
```diff
@@ -416,16 +417,40 @@ extension ConversationRequestStrategy: ZMUpstreamTranscoder {
...
+        ) {
             managedObjectContext.delete(existingConversation)
             deletedDuplicate = true
         }
 
         newConversation.remoteIdentifier = conversationID
         payload.updateOrCreate(in: managedObjectContext)
         newConversation.needsToBeUpdatedFromBackend = deletedDuplicate
+
+        if newConversation.messageProtocol == .mls {
```
Comment (David-Henner) [inline Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift:434]: When creating the mls conversation, we pass `nil` users in the `NewConversation` payload. Which means that the conversation we get as a result (`newConversation`) won't have any users except from the self user. As a result of this, we won't claim key packages nor send any welcome / handshake messages to the conversation members.

Comment (johnxnguyen) [reply Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift:434]: yep I noticed this too... not sure what's the best way to work around this.

File: Sources/Payloads/Processing/PayloadProcessing+Conversation.swift (hunk 1 above)
Comment (johnxnguyen) [reply Sources/Payloads/Processing/PayloadProcessing+Conversation.swift:171]: removed

File: Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift (hunk 2 above)
Comment (johnxnguyen) [reply Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift:442]: I'll address this in another PR

**Pull request merged**